### PR TITLE
feat: merge multi-calendar support from alpha to master

### PR DESCRIPTION
**Pull request by kabaros**
This PR includes the updates to alpha channel to use the multi-calendar-dates library to generate fixed periods in order to support non-Gregorian calendrical systems (such as Nepali and Ethiopic).

Implements [DHIS2-14709](https://dhis2.atlassian.net/browse/DHIS2-14709). **Used in this Data-Visualiser PR: https://github.com/dhis2/data-visualizer-app/pull/2233**

### Notes

- The previous implementation came with an [comprehensive set of snapshot tests](https://github.com/dhis2/analytics/blob/master/src/components/PeriodDimension/__tests__/__snapshots__/fixedPeriods.spec.js.snap) that cover all period generation scenarios and edge cases (especially with EPI week generation). This test suite guided this new implementation, to ensure that all the subtle edge cases were covered.

- The bulk of these changes were reviewed by Edoardo, Abyot and Hendrik in [this PR](https://github.com/dhis2/analytics/pull/1407). The description and discussions there can provide more context for the changes.

### Testing

- This can be tested from the [Data-Visualiser PR deploy link](https://github.com/dhis2/data-visualizer-app/pull/2233#issuecomment-1415685112) which uses the alpha release. For the purpose of merging this PR, the testing should concentrate on testing the Gregorian calendar fixed periods in Data-Visualiser and ensure that it still has the same behaviour as before.

- [optional] Testing for other calendrical systems (ethiopic and nepali) can be done as well, but only for the FE, as we don't have a readily-available DHIS2 instance configured with these calendars.

---

### Key features

1. Generate fixed periods in multiple calendrical systems (i.e. Gregorian, Nepali, Ethiopic etc...)
2. Get the current date in multiple calendars
3. Consume and use the currently configured calendar in DHIS2
4. Localise periods using browsers' Intl.DateTimeFormat

---

### BREAKING CHANGE

No breaking change. The existing period generation logic in analytics came with an exhaustive suite of tests. The library used those as a specs to guide the implementation, and make sure they all pass, so the library should not change the existing behaviour, only extend it to support other calendars ✅ 

---

### Screenshots

https://user-images.githubusercontent.com/1014725/220880611-133e7208-596d-4d48-944a-03ac7459e068.mov

[DHIS2-14709]: https://dhis2.atlassian.net/browse/DHIS2-14709?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by HendrikThePendric — APPROVED** [review]

**Review by janhenrikoverland — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -6980,6 +6996,11 @@ classnames@^2.2.5, classnames@^2.2.6, classnames@^2.3.1:
   resolved "https://registry.yarnpkg.com/classnames/-/classnames-2.3.1.tgz#dfcfa3891e306ec1dad105d0e88f4417b8535e8e"
   integrity sha512-OlQdbZ7gLfGarSqxesMesDa5uz7KFbID8Kpq/SxIoNGDqY8lSYs0D+hhtBXhcdB3rcbXArFr7vlHheLk1voeNA==
 
+classnames@^2.3.2:
+  version "2.3.2"
```
Comment (martinkrulltott) [inline yarn.lock:7000]: This is a duplicate of `classnames@^2.2.5, classnames@^2.2.6, classnames@^2.3.1` above. 
Both should resolve to `2.3.2`

Comment (kabaros) [reply yarn.lock:7000]: I updated it [here](https://github.com/dhis2/analytics/pull/1436) since this PR was already merged.
